doc/releases/migration-guide-3.7.rst

```diff
@@ -22,6 +22,36 @@ Build System
   out-of-tree SoCs and boards to be ported to the new model. See the
   :ref:`hw_model_v2` for more detailed information. (:github:`69607`)
 
+* The following build-time generated headers:
+
+  .. list-table::
+     :header-rows: 1
+
+     * - Affected header files
+     * - ``app_version``
+     * - ``autoconf``
+     * - ``cmake_intdef``
+     * - ``core-isa-dM``
+     * - ``devicetree_generated``
+     * - ``driver-validation``
+     * - ``kobj-types-enum``
+     * - ``linker-kobject-prebuilt-data``
+     * - ``mcuboot_version``
+     * - ``offsets``
+     * - ``otype-to-size``
+     * - ``otype-to-str``
+     * - ``strerror_table``
+     * - ``strsignal_table``
+     * - ``syscall_list``
+     * - ``version``
+     * - ``zsr``
```

Comment:
Should these have a `.h` postfix?

Reply:
The omission is actually intentional (implied by them being header files) but I can add them if you think it increases clarity

Reply:
Fixed